Example/MoyaComparison/MoyaComparison/Services/DataStores/VideoDataStore.swift

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright 2017 niji
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import RxSwift
+
+struct VideoDataStore {
+  
+  weak var delegate: VideoView?
+  
+  private var disposeBag = DisposeBag()
+  private var webService: VideoWebService?
+  
+  init(webService: VideoWebService?) {
+    self.webService = webService
+  }
+  
+  func fetchVideos() {
+    self.webService?.fetchAllVideos()
+      .subscribeOn(ConcurrentDispatchQueueScheduler(qos: .background))
+      .observeOn(MainScheduler.instance)
+      .do(
+        onNext: { (videos: [Video]) in
+          self.delegate?.fetched(videos: videos)
+          logger.debug(.webServiceClient, "received videos : \(videos)")
+      },
+        onError: { (error: Error) in
+          let message = "Failed to fetch videos with error : \(error)"
+          self.delegate?.error(message: message)
+          logger.error(.webServiceClient, message)
+      }, onCompleted: nil, onSubscribe: nil, onSubscribed: nil, onDispose: nil)
+      .subscribe()
+      .addDisposableTo(self.disposeBag)
+  }
+  
+  func addVideo() {
+    guard let video = webService?.fakeVideoToAdd() else {
+      logger.error(.webServiceClient, "Failed to create fake video !")
+      return
+    }
+    self.webService?.addVideo(video: video)
+      .subscribeOn(ConcurrentDispatchQueueScheduler(qos: .background))
+      .observeOn(MainScheduler.instance)
+      .do(onNext: { (video: Video) in
+        self.delegate?.added(video: video)
+      }, onError: { (error: Error) in
+        let message = "Failed to add video with error : \(error)"
+        self.delegate?.error(message: message)
+        logger.error(.webServiceClient, message)
+      }, onCompleted: nil, onSubscribe: nil, onSubscribed: nil, onDispose: nil)
+      .subscribe()
+      .addDisposableTo(self.disposeBag)
+  }
+  
+  func deleteVideo(identifier: Int) {
+    self.webService?.deleteVideo(identifier: identifier)
+      .subscribeOn(ConcurrentDispatchQueueScheduler(qos: .background))
+      .observeOn(MainScheduler.instance)
+      .do(onNext: nil, onError: { (error: Error) in
```

Comment:
(Of course all those remarks above apply to everywhere you do a similar Rx call chain in your demo app, I didn't want to repeat the comments everywhere)